browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -128,7 +168,28 @@ BraveShieldsActionView::CreateDefaultBorder() const {
 }
 
 void BraveShieldsActionView::Update() {
-  // We can get active webcontent's url and perform a GetBraveShieldsEnabled
-  // check
-  NOTIMPLEMENTED();
+  UpdateIconState();
+}
+
+void BraveShieldsActionView::OnResourcesCountChange(const int count) {
+  UpdateIconState();
+}
+
+void BraveShieldsActionView::OnTabStripModelChanged(
+    TabStripModel* tab_strip_model,
+    const TabStripModelChange& change,
+    const TabStripSelectionChange& selection) {
+  if (selection.active_tab_changed()) {
+    if (selection.new_contents) {
+      brave_shields::BraveShieldsDataController::FromWebContents(
```

Comment:
why does this only get set as an observer when the tab strip model changes? Also it doesn't appear to be removed if it does get added

Reply:
adding a check for change kRemoved tab